Core/Python/add_device_to_static_group.py

```diff
@@ -20,33 +20,158 @@
 #
 """
 SYNOPSIS:
-   Add a host to a static group
+   Add one or more hosts to an existing static group.
 
 DESCRIPTION:
-   This script exercises the OME REST API to create a new static
-   group. The user is responsible for adding devices to the
-   group once the group has been successfully created.
-   For authentication X-Auth is used over Basic Authentication
-   Note that the credentials entered are not stored to disk.
+   This script exercises the OME REST API to add one or more
+   hosts to an existing static group. For authentication X-Auth
+   is used over Basic Authentication. Note: The credentials entered
+   are not stored to disk.
 
 EXAMPLE:
-   python create_static_group.py --ip <xx> --user <username>
-        --password <pwd> --groupname "Random Test Group" --devicename "cmc1"
+   python add_device_to_static_group.py --ip <xx> --user <username>
+        --password <pwd> --groupname "Random Test Group" --devicenames "cmc1,host3,192.168.1.5"
 """
 
 import json
-import sys
 import argparse
 from argparse import RawTextHelpFormatter
 import urllib3
 import requests
 
 
-def add_device_to_static_group(ome_ip_address, ome_username, ome_password, group_name):
-    """ Authenticate with OME and enumerate groups """
+def get_device_list(ome_ip_address: str, headers: dict) -> list:
+    """
+    Retrieves a list of all devices being handled by this OME server
+
+    Args:
+        ome_ip_address: The IP address of the OME server
+        headers: Headers used for authentication to the OME server
+
+    Returns: A list of all devices managed by the this OME server
+
+    """
+
+    print("Retrieving a list of all devices...")
+    next_link_url = "https://%s/api/DeviceService/Devices" % ome_ip_address
+    device_data = None
+
+    while next_link_url is not None:
+        device_response = requests.get(next_link_url, headers=headers, verify=False)
+        next_link_url = None
+        if device_response.status_code == 200:
+            data = device_response.json()
+            if data['@odata.count'] <= 0:
+                print("No devices are managed by OME server: " + ome_ip_address + ". Exiting.")
+                exit(1)
```

Comment:
This isn't a required change but my personal preference is to have exit points from the main method (add_device_to_static_group) and return invalid values / None from these utility methods

Reply:
Updated